libbeat/outputs/elasticsearch/client.go

```diff
@@ -185,6 +186,36 @@ func NewClient(
 		proxyURL:         s.Proxy,
 	}
 
+	if s.CacheRedirect {
+		client.Connection.http.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+			if len(via) != 1 {
+				return nil
+			}
+
+			if via[0].Method != "GET" {
+				return http.ErrUseLastResponse
+			}
+
+			newURL := req.URL.String()
+
+			logp.Info("Caching ES HTTP API Redirect: %s", newURL)
+
+			s.URL = newURL
+
+			newClient, err := NewClient(s, onConnectCallback)
+
+			if err != nil {
+				logp.Err("Failed to Cache ES HTTP API Redirect to %s: %s", newURL, err)
+				return http.ErrUseLastResponse
+			}
+
+			client.bulkRequ = newClient.bulkRequ
+			client.Connection = newClient.Connection
+
+			return http.ErrUseLastResponse
```

Comment:
same, using `ErrUseLastResponse` can lead to data-loss (which should be fixed, yeah)... I wonder if we want to enforce an error here (enforce client to ping endpoint before trying to send data) or just continue with redirect...